docs/observability/aws/deploy-use-aws-observability/deploy-with-terraform.md

```diff
@@ -1547,3 +1547,53 @@ The following table provides a list of all source parameters and their default v
 | `email_notifications`	Email Notifications to be sent by the alert. | `[ ]` |
 | `connection_notifications` | Connection Notifications to be sent by the alert. | `[ ]` |
 | `parent_folder_id` | The folder ID is automatically generated. Do not enter a value for this parameter. This is the folder ID to install the apps into. A folder using the provided name will be added in "apps_folder_name". If the folder ID is empty, apps will be installed in the Personal folder. | Ignore this parameter. |
+
+## Troubleshooting
+
+This section provides information on how to troubleshoot failures while deploying AWS observability solution using Terraform.
+
+### 1. Python command not found
+#### Error Message
+`python source-module/attach_fields_to_source.py`<br/>Python: command not found
+#### Solution
+Identify and replace `python` with `python3` in [source-module/update_sources.tf](https://github.com/SumoLogic/sumologic-solution-templates/blob/AWSO_FY23Q4_Release/aws-observability-terraform/source-module/update_sources.tf#L12).
+
+### 2. Module not found
+#### Error Message
+`Local-exec provisioner error`<br/>Module Not Found Error: No Module named ‘sumologic’
+#### Solution
+Verify you configured [Sumo Logic provider](https://github.com/SumoLogic/sumologic-solution-templates/blob/AWSO_FY23Q4_Release/aws-observability-terraform/providers.tf#L1).
+
+### 3. Hierarchy named 'AWS Observability' already exist
+#### Error Message
+`"errors":[{"code":"hierarchy:duplicate","message":"hierarchy named 'AWS Observability' already exist"}]`
+#### Solution
+Delete existing hierarchy and a create new one:<br/>
+1. Get Hierarchy-id list of existing hierarchies and keep it noted.<br/>
+  `curl -s -H 'Content-Type: application/json' --user <accessid>:<accesskey> -X GET https://<apiendpoint>/api/v1/entities/hierarchies`
+1. Delete the existing Hierarchy.<br/>
+  `curl -s -H 'Content-Type: application/json' --user <accessid>:<accesskey> -X DELETE https://<apiendpoint>/api/v1/entities/hierarchies/<hierarchyid>`<br/>Learn [more](https://help.sumologic.com/docs/api/getting-started/#sumo-logic-endpoints-by-deployment-and-firewall-security) for apiendpoint.
+
+### 4. Cannot import name 'SumoLogic' from 'sumologic'
+#### Error Message
+from sumologic import SumoLogic<br/>Import Error: cannot import name 'SumoLogic' from 'sumologic'<br/> `(/usr/local/lib/python3.10/site-packages/sumologic/__init__.py)`
+#### Solution
+The package is [sumologic-sdk](https://pypi.org/project/sumologic-sdk/) and install it for AWS observability solution using the following command:<br/>`pip install sumologic-sdk`
```

Comment:
Format this as [code block](https://help.sumologic.com/docs/contributing/style-guide/#code-blocks) (not code font) because it's a snippet we want customers to copy/paste

Reply:
Comment incorporated.